homeassistant/components/hdmi_cec.py

```diff
@@ -4,118 +4,535 @@
 For more details about this component, please refer to the documentation at
 https://home-assistant.io/components/hdmi_cec/
 """
+import asyncio
 import logging
+import threading
+import time
 
+import cec
+import os
 import voluptuous as vol
+from collections import defaultdict
+from functools import reduce
+from homeassistant.components import discovery
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (EVENT_HOMEASSISTANT_START, STATE_ON, STATE_OFF,
+                                 STATE_UNKNOWN, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
 
-from homeassistant.const import (EVENT_HOMEASSISTANT_START, CONF_DEVICES)
-import homeassistant.helpers.config_validation as cv
+DOMAIN = 'hdmi_cec'
 
-_CEC = None
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_DEVICE = 'device'
+CEC_CLIENT = {}
 
-DOMAIN = 'hdmi_cec'
+SCAN_INTERVAL = 30
 
-MAX_DEPTH = 4
+VENDORS = {0x000039: 'Toshiba',
```

Comment:
There is too much protocol specific logic in this component. This should be moved to a third party lib on which Home Assistant can depend.